src/components/v5/common/ActionSidebar/partials/forms/StagedPaymentForm/hooks.ts

```diff
@@ -0,0 +1,189 @@
+import { Id } from '@colony/colony-js';
+import { useMemo } from 'react';
+import { type DeepPartial } from 'utility-types';
+import { type InferType, array, number, object, string } from 'yup';
+
+import { MAX_MILESTONE_LENGTH } from '~constants';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import useNetworkInverseFee from '~hooks/useNetworkInverseFee.ts';
+import { ActionTypes } from '~redux';
+import { mapPayload } from '~utils/actions.ts';
+import { notNull } from '~utils/arrays/index.ts';
+import getLastIndexFromPath from '~utils/getLastIndexFromPath.ts';
+import { formatText } from '~utils/intl.ts';
+import { amountGreaterThanZeroValidation } from '~utils/validation/amountGreaterThanZeroValidation.ts';
+import {
+  ACTION_BASE_VALIDATION_SCHEMA,
+  DECISION_METHOD_FIELD_NAME,
+  AMOUNT_FIELD_NAME,
+  TOKEN_FIELD_NAME,
+  CREATED_IN_FIELD_NAME,
+  RECIPIENT_FIELD_NAME,
+  FROM_FIELD_NAME,
+} from '~v5/common/ActionSidebar/consts.ts';
+import useActionFormBaseHook from '~v5/common/ActionSidebar/hooks/useActionFormBaseHook.ts';
+import { type ActionFormBaseProps } from '~v5/common/ActionSidebar/types.ts';
+
+import { allTokensAmountValidation } from '../PaymentBuilderForm/utils.tsx';
+
+import { getStagedPaymentPayload } from './utils.ts';
+
+export const useValidationSchema = (networkInverseFee: string | undefined) => {
+  const { colony } = useColonyContext();
+  const colonyTokens = useMemo(
+    () =>
+      colony.tokens?.items
+        .filter(notNull)
+        .map((colonyToken) => colonyToken.token) || [],
+    [colony.tokens?.items],
+  );
+
+  return useMemo(
+    () =>
+      object()
+        .shape({
+          [RECIPIENT_FIELD_NAME]: string()
+            .required(({ path }) => {
+              const index = getLastIndexFromPath(path);
+              if (index === undefined) {
+                return formatText({ id: 'errors.amount' });
```

Comment:
This error doesn't seem right for the recipient field validation

Reply:
And does the recipient field need any of the index checking code considering there's only 1 per payment?